File: group-projects/2021-03_media_bazaar/MediaBazaar Website/classes/EmployeeWorkday.php
<?php
class EmployeeWorkday extends Database {
    private $day_id;
    private $employee_id;
    private $first_shift;
    private $second_shift;
    private $absence;
    private $absence_reason;

    public function __construct($day_id, $employee_id, $first_shift, $second_shift, $absence, $absence_reason) {
        $this->day_id = $day_id;
        $this->employee_id = $employee_id;
        $this->first_shift = $first_shift;
        $this->second_shift = $second_shift;
        $this->absence = $absence;
        $this->absence_reason = $absence_reason;
    }

    public function getDayId() {
        return $this->day_id;
    }

    public function getFirstShift() {
        return $this->first_shift;
    }

    public function getSecondShift() {
        return $this->second_shift;
    }

    public function getShiftsString() {
        $str = "";

        if(strtolower($this->first_shift) != 'none') {
            $str .= $this->first_shift;
        }

        if(strtolower($this->second_shift) != 'none') {
            $str .= " & " . $this->second_shift;
        }

        return $str;
    }
}

File: group-projects/2021-03_media_bazaar/MediaBazaar - auto data algorithm/PRJMediaBazaar/Logic/EmployeeWorkday.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PRJMediaBazaar.Logic
{
    class EmployeeWorkday
    {
        //public EmployeeWorkday(Employee emp, Shift firstShift, Shift secondShift, bool absence, AbsenceReason absenceReason)
        //{
        //    Employee = emp;
        //    FirstShift = firstShift;
        //    SecondShift = secondShift;
        //    Absence = absence;
        //    AbsenceReason = AbsenceReason;
        //}

        public EmployeeWorkday(int dayId,Employee emp, Shift firstShift, Shift secondShift, bool absence, AbsenceReason absenceReason, double hours)
        {
            Employee = emp;
            FirstShift = firstShift;
            SecondShift = secondShift;
            Absence = absence;
            AbsenceReason = absenceReason;
            Hours = hours;
            DayId = dayId;
        }
        public EmployeeWorkday(int dayId, Employee emp, Shift firstShift, Shift secondShift, bool absence, AbsenceReason absenceReason)
        {
            Employee = emp;
            FirstShift = firstShift;
            SecondShift = secondShift;
            Absence = absence;
            AbsenceReason = absenceReason;
            DayId = dayId;
        }


        public Employee Employee { get; private set; }
        public Shift FirstShift { get; set; }
        public Shift SecondShift { get; set; }
        public bool Absence { get; set; }
        public AbsenceReason AbsenceReason { get; set; }
        public int DayId { get; private set; }

        public double Hours { get; private set; }

        public string GetOccupation()
        {
            if (FirstShift.ToString() == "None" && SecondShift.ToString() != "None")
            {
                return SecondShift.ToString();
            }
            else if (SecondShift.ToString() == "None" && FirstShift.ToString() != "None")
            {
                return FirstShift.ToString();
            }
            return "Double Shift";
        }

        public string GetBusyShift()
        {
            if (FirstShift.ToString() == "None" && SecondShift.ToString() != "None")
            {
                return SecondShift.ToString();
            }
            else if (SecondShift.ToString() == "None" && FirstShift.ToString() != "None")
            {
                return FirstShift.ToString();
            }
            return "-1";
        }

        public string GetShift()
        {
            if(FirstShift.ToString() == "None")
            {
                return SecondShift.ToString();
            }
            return FirstShift.ToString();
        }
    }
}
